### PR TITLE
Reactivate Sassdoc, clean up docs build

### PR DESCRIPTION
**Pull request by davidknezic**
This reintroduces SassDoc, after the fix of interpolated selectors.

screen shot 2016-10-31 at 09 18 31

However, our API section currently looks very chaotic and isn't probably easy to use as a developer reference.

How about we display the items *Classes*, *CSS Utilities*, *Mixins* and *jQuery Plugins* under the *API* menu item? Then each of these items is a very long page containing all of the corresponding docs. This allows for a quick search using the browser-search.

### CODE REVIEW

**Comment by elessar-ch** [comment]
We could even put all the classes utilities and so on into one page. Might be a bit long though.
A page-navigation with anchors might be good in either case.

**Comment by davidknezic** [comment]
@elessar-ch do you have any clue, why the example doesn't get rendered properly in the image above? Also, is it normal that there's that weird indentation in the source code?

**Comment by FKSI** [comment]
+1 with @elessar-ch I'd love to have all utilities on a single page even thought it might be a little long.

**Comment by davidknezic** [comment]
Related https://github.com/SassDoc/scss-comment-parser/pull/25

**Comment by davidknezic** [comment]
Ready to review. I think the fix for SassDoc can be done at a later point, I'll add an issue.

**Comment by davidknezic** [comment]
This pr also remove a LOT of old code for pug and markdown parsing. There were also lots of unneeded dependencies.

File: lib/build-docs.js [hunk 1]
```diff
@@ -269,7 +204,10 @@ metalsmith.destination('./dist/docs/')
 // Function to run the magic
 metalsmith.build((err) => {
   if (err) {
+    // eslint-disable-next-line no-console
     console.log((new Date()).toLocaleTimeString(), 'Failed!', err)
```
Comment (FKSI) [inline lib/build-docs.js]: How about switching to console.err() aren't they made for that?

**Review by FKSI — CHANGES_REQUESTED** [review]
Try faster

**Review by FKSI — APPROVED** [review]

**Pull request merged**